compiler/rustc_codegen_llvm/src/asm.rs

```diff
@@ -1020,6 +1020,26 @@ fn llvm_fixup_input<'ll, 'tcx>(
                 value
             }
         }
+        (
+            InlineAsmRegClass::Arm(
+                ArmInlineAsmRegClass::dreg
+                | ArmInlineAsmRegClass::dreg_low8
+                | ArmInlineAsmRegClass::dreg_low16,
+            ),
+            Abi::Vector { element, count: 4 },
+        ) if element.primitive() == Primitive::Float(Float::F16) => {
+            bx.bitcast(value, bx.type_f64())
```

Comment:
Can you transmute it to `i16x4` instead? I'm concerned about potential issues on big-endian targets.

Reply:
In that case we should move all vector types to be under the `neon` feature rather than `vfp2`, otherwise the exact same problem will happen for `i16x4`.